caluma-demo/app/routes/application.js

```diff
@@ -1,12 +1,44 @@
 import { inject as service } from "@ember/service";
 import Route from "@ember/routing/route";
+import config from "../config/environment";
 
+const { languages, fallbackLanguage } = config;
 export default class ApplicationRoute extends Route {
   @service intl;
 
-  beforeModel() {
-    super.beforeModel(...arguments);
+  languages;
+  fallbackLanguage;
 
-    return this.intl.setLocale("en");
+  getBrowserLanguage() {
+    const preferred = (navigator.languages || [navigator.language]).map(
+      locale => locale
+    );
+
+    return preferred.find(lang => languages.includes(lang));
+  }
+
+  getLocalStorageLanguage() {
+    if (languages.includes(localStorage.getItem("language"))) {
+      return localStorage.getItem("language");
+    }
+  }
+
+  getLanguage(queryParamLanguage) {
+    if (languages.includes(queryParamLanguage)) {
```

Comment:
Is the `queryParamLanguage` part really needed?

Reply:
No it's not needed. I've removed it.